src/collection.ts

```diff
@@ -217,29 +218,57 @@ export abstract class ReadOnlyCollection<T extends TigrisCollectionType> impleme
 	 *
 	 * @param callback - Callback to consume events asynchronously
 	 */
-	events(callback: EventsCallback<T>) {
+	events(): Observable<StreamEvent<T>>;
+	events(callback: EventsCallback<T>): void;
+	events(callback?: EventsCallback<T>): Observable<StreamEvent<T>> {
 		const eventsRequest = new ProtoEventsRequest()
 			.setDb(this.db)
 			.setCollection(this.collectionName);
 
 		const stream: grpc.ClientReadableStream<ProtoEventsResponse> =
 			this.grpcClient.events(eventsRequest);
 
-		stream.on("data", (eventsResponse: ProtoEventsResponse) => {
-			const event = eventsResponse.getEvent();
-			callback.onNext(
-				new StreamEvent<T>(
-					event.getTxId_asB64(),
-					event.getCollection(),
-					event.getOp(),
-					Utility.jsonStringToObj<T>(Utility._base64Decode(event.getData_asB64()), this.config),
-					event.getLast()
-				)
-			);
-		});
+		if (callback !== undefined) {
```

Comment:
Do you think it will be useful to provide an observer in [Cursor](https://github.com/tigrisdata/tigris-client-ts/blob/main/src/consumables/abstract-cursor.ts). Similar to `find()` this `events()` api can return a Cursor and consume it as a Streamable, iterable or ab Observable. What do you think?

Is this RxJS in general used for consumption on client side? Or also on server side applications?

Reply:
Good question, I am not sure how is the `RxJS` consumption on server side applications but I would think it is used. Specially for people with Angular background would like to see it.

As for Cursor approach, I think RxJS is an alternate to it. It would make API complex if add it to Cursor API.

Reply:
I just learnt from the team that this Change events support has been removed from the server and this API probably won't be working. I'll remove this from client now

Reply:
We will be reimplementing real-time streaming using websockets